### PR TITLE
feat: use worker bindings interacting from api to gateway

### PR DESCRIPTION
**Pull request by vasco-santos**
This PR adds:
- worker bindings from API to Gateway

Implementation details:
- miniflare support for testing bindings is still quite limited, we cannot spin up the other worker build, and it does not seem to support ESM either...
- bindings from API to Gateway tests use custom script that does the fetch to the local daemon converting subdomain resolution to path resolution. This is because we can't resolve localhost subdomains locally unless we add them to `/etc/hosts`, which is not cool, given we can't use tools like https://github.com/feross/hostile without sudo for pre and post tests... There was the option of relying on services like https://readme.localtest.me/ but don't really want to use external services in CI

Needs:
- [x] wrangler 2

Initial binding part of https://github.com/nftstorage/nftstorage.link/issues/101 to move to worker bindings setup

### CODE REVIEW

File: packages/api/wrangler.toml [hunk 1]
```diff
@@ -21,12 +21,17 @@ NFT_STORAGE_API = "https://api.nft.storage"
 DEBUG = "false"
 ENV = "production"
 GATEWAY_DOMAIN = "nftstorage.link"
-IPFS_GATEWAYS = "[\"https://ipfs.io\", \"https://cf.nftstorage.link\", \"https://nft-storage.mypinata.cloud/\"]"
 
 [[env.production.r2_buckets]]
 bucket_name = "super-hot"
 binding = "SUPERHOT"
 
+[[env.production.unsafe.bindings]]
```
Comment (olizilla) [inline packages/api/wrangler.toml:29]: are `unsafe.bindings` documented anywhere? Can you link to docs or add a note here about how they work and why they may or may not be unsafe?

Comment (olizilla) [reply packages/api/wrangler.toml:29]: All i can find is this PR which mentions they are undocumented https://github.com/cloudflare/wrangler2/issues/1004

Comment (olizilla) [reply packages/api/wrangler.toml:29]: and this... the list of drawbacks listed on this PR is scary
https://github.com/cloudflare/wrangler2/pull/906

File: packages/api/test/scripts/gateway.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  },
+}
+
+/**
+ * Parse subdomain URL and return cid
+ *
+ * @param {URL} url
+ */
+function getCidFromSubdomainUrl(url) {
+  // Replace "ipfs-staging" by "ipfs" if needed
```
Comment (olizilla) [inline packages/api/test/scripts/gateway.js:25]: Comment looks like a copy-pasta... or is there a deployment issue i'm not seeing
```suggestion
```

File: packages/api/wrangler.toml [hunk 3]
```diff
@@ -21,12 +21,17 @@ NFT_STORAGE_API = "https://api.nft.storage"
...
+name = "GATEWAY"
+type = "service"
+service = "gateway-nft-storage-production"
```
Comment (olizilla) [inline packages/api/wrangler.toml:32]: do we have a naming scheme for our services? Should we define one?

**Review by olizilla — COMMENTED** [review]
The code changes look good, but I need a bit more context on what you know so far about this "app-specific-gateway worker to shared-gateway worker" pattern will get us. 

Will it really get us a common cache? Or do we need to test it to find out?

If a bound worker is really just code bundled with another worker, I would guess that we still wouldn't benefit from a shared common cache, but each app-gateway-specific-gateway would just be writiing to it's own cache just using the shared code of the shared-gateway.

File: packages/api/wrangler.toml (hunk 1 above)
Comment (vasco-santos) [reply packages/api/wrangler.toml:29]: The unsafe is not needed anymore in fact. They already support without it :)

The drawbacks there seem outdated by know from what I have seen in their discord

**Comment by vasco-santos** [comment]
> Will it really get us a common cache? Or do we need to test it to find out?

Let's talk sync on this. btw, this is not related at all with this PR 😛 This is basically using bindings between nftstorage.link gateway and API (for perma-cache)

**Review by olizilla — APPROVED** [review]
LGTM

**Pull request merged**